src/Nethermind/Nethermind.AuRa.Test/Validators/ReportingContractBasedValidatorTests.cs

```diff
@@ -136,13 +137,34 @@ public void Adds_transactions_to_block([Values(0, 5, 15)] int validatorsToReport
         public void Reports_skipped_blocks()
         {
             var context = new TestContext(false, initialValidators: new[] {TestItem.AddressA, NodeAddress, TestItem.AddressC, TestItem.AddressD});
+            context.ReportingValidatorContract.ReportBenign(Arg.Any<Address>(), Arg.Any<UInt256>()).Returns(new GeneratedTransaction());
             var parent = Build.A.BlockHeader.WithNumber(10).WithAura(10).TestObject;
             var header = Build.A.BlockHeader.WithParent(parent).WithAura(20).TestObject;
             context.Validator.TryReportSkipped(header, parent);
             context.ReportingValidatorContract.Received(1).ReportBenign(TestItem.AddressA, (UInt256) header.Number);
             context.ReportingValidatorContract.Received(1).ReportBenign(TestItem.AddressC, (UInt256) header.Number);
             context.ReportingValidatorContract.Received(1).ReportBenign(TestItem.AddressD, (UInt256) header.Number);
             context.ReportingValidatorContract.Received(0).ReportBenign(NodeAddress, (UInt256) header.Number);
+            context.TxSender.Received(3).SendTransaction(Arg.Is<Transaction>(t => t is GeneratedTransaction), TxHandlingOptions.ManagedNonce);
+        }
+        
+        [Test]
+        public void Report_ignores_duplicates_in_same_block()
+        {
+            var context = new TestContext(true, initialValidators: new[] {TestItem.AddressA, NodeAddress, TestItem.AddressC});
+            var transaction = Build.A.Transaction.TestObject;
+            context.ReportingValidatorContract.ReportBenign(Arg.Any<Address>(), Arg.Any<UInt256>()).Returns(transaction); 
+            context.ReportingValidatorContract.ReportMalicious(Arg.Any<Address>(), Arg.Any<UInt256>(), Arg.Any<byte[]>()).Returns(transaction);
+            
+            context.Validator.ReportBenign(MaliciousMinerAddress, 100, IReportingValidator.BenignCause.FutureBlock); // sent
+            context.Validator.ReportBenign(MaliciousMinerAddress, 100, IReportingValidator.BenignCause.IncorrectProposer); // sent
+            context.Validator.ReportBenign(MaliciousMinerAddress, 100, IReportingValidator.BenignCause.FutureBlock); // ignored
+            context.Validator.ReportMalicious(MaliciousMinerAddress, 100, Bytes.Empty, IReportingValidator.MaliciousCause.DuplicateStep); // sent
```

Comment:
shall we test all potential combinations?

Reply:
Adding some more...